dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,561 @@
+                }
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnElementIndexChangedEvent(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementIndexChangedEventArgs& args)
+{
+    if (auto const element = args.Element())
+    {
+        element.SetValue(winrt::AutomationProperties::PositionInSetProperty(), box_value(args.NewIndex() + 1));
+
+        // When the selected item's index changes, update selection to match
+        if (auto const elementAsToggle = element.try_as<winrt::ToggleButton>())
+        {
+            if (SharedHelpers::IsTrue(elementAsToggle.IsChecked()))
+            {
+                FocusElement(args.NewIndex());
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnElementClearingEvent(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementClearingEventArgs& args)
+{
+    if (const auto& item = args.Element().try_as<winrt::BreadcrumbItem>())
+    {
+        const auto& itemImpl = winrt::get_self<BreadcrumbItem>(item);
+        itemImpl->ResetVisualProperties();
+        itemImpl->RevokeListeners();
+    }
+}
+
+winrt::IInspectable Breadcrumb::GenerateInternalItemsSource()
+{
+    // An internal ItemsSource is generated internally to add the
+    // ellipsis button
+    const auto& newItemsSource = winrt::make<Vector<IInspectable>>();
+
+    // The ellipsis button is created as a BreadcrumbItem, the element
+    // factory verifies that this element is already a BreadcrumbItem
+    // so no extra processing is needed
+    const auto& ellipsisItem = winrt::make<BreadcrumbItem>();
+    ellipsisItem.Content(winrt::box_value(L"..."));
+    newItemsSource.Append(ellipsisItem);
+
+    if (const auto& itemsList = ItemsSource().try_as<winrt::Foundation::Collections::IIterable<winrt::IInspectable>>())
+    {
+        for (const auto& item : itemsList)
+        {
+            newItemsSource.Append(item);
+        }
+    }
+
+    return newItemsSource;
+}
+
+void Breadcrumb::RaiseItemClickedEvent(const winrt::IInspectable& content)
+{
+    const auto& eventArgs = winrt::make_self<BreadcrumbItemClickedEventArgs>();
+    eventArgs->Item(content);
+    m_itemClickedEventSource(*this, *eventArgs);
+}
+
+winrt::Collections::IVector<winrt::IInspectable> Breadcrumb::HiddenElements()
+{
+    // The hidden element list is generated in the BreadcrumbLayout during
+    // the arrange method, so we retrieve the list from it
+    if (const auto& breadcrumbItemRepeater = m_breadcrumbItemRepeater.get())
+    {
+        if (const auto& breadcrumbLayout = breadcrumbItemRepeater.Layout().try_as<BreadcrumbLayout>())
+        {
+            return breadcrumbLayout->HiddenElements();
+        }
+    }
+
+    // By default just return an empty list
+    return winrt::make<Vector<winrt::IInspectable>>();
+}
+
+// When focus comes from outside the Breadcrumb control we will put focus on the selected item.
+void Breadcrumb::OnGettingFocus(const winrt::IInspectable&, const winrt::GettingFocusEventArgs& args)
```

Comment:
Does this method work when Breadcrumb.ItemTemplate contains a bunch of focusable Control instances?